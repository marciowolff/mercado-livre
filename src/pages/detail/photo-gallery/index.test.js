import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import PhotoGallery from './index'
import { wrap } from 'module'
let wrapper

it('PhotoGallery should be a function', () => {
  expect(typeof PhotoGallery).toBe('function')
})

it('Should PhotoGallery to match snapshot', () => {
  const tree = renderer.create(<PhotoGallery items={[ {url: 'url-1'}, { url: 'url-2' } ]}/>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should render PhotoGallery without items', () => {
  const wrapper = mount(<PhotoGallery  />)

  expect(wrapper.find('.photoGallery')).toHaveLength(0)
})

it('Should render PhotoGallery update tag element', () => {
  const wrapper = mount(<PhotoGallery tag='aside' items={[{url: 'url-1'}]} />)

  expect(wrapper.find('aside.photoGallery')).toHaveLength(1)
})

describe('render', () => {
  beforeEach(() => {
    wrapper = mount(<PhotoGallery items={[ {url: 'url-1'}, { url: 'url-2' } ]} />)
  })

  it('should render photoGallery', () => {
    expect(wrapper.find('.photoGallery')).toBeTruthy()
  })

  it('should render img featured equal first item', () => {
    expect(wrapper.find('figure img').props().src).toEqual('url-1')
  })

  it('should render thumbs', () => {
    const thumbs = wrapper.find('.thumbs img')

    expect(thumbs).toHaveLength(2)
  })

  it('event click thumbs update featured img', () => {
    expect(wrapper.find('figure img').props().src).toEqual('url-1')

    wrapper.find('.thumbs img').at(1).simulate('click')

    expect(wrapper.find('figure img').props().src).toEqual('url-2')
  })
})
