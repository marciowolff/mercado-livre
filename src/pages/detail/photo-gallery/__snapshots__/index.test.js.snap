// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should PhotoGallery to match snapshot 1`] = `
<div
  className="photoGallery "
>
  <figure>
    <img
      src="url-1"
    />
  </figure>
  <span
    className="thumbs"
  >
    <img
      className="active"
      onClick={[Function]}
      src="url-1"
    />
    <img
      className=""
      onClick={[Function]}
      src="url-2"
    />
  </span>
</div>
`;
